### PR TITLE
cmd/syncthing, gui: Improve completion calculation (fixes #3492)

### LINKED ISSUE #3492: gui: Completion broken
So this breaks the UI, due to a race condition.

```
logbar.js:11 TypeError: Cannot read property 'globalBytes' of undefined
    at syncthingController.js:479
    at angular.js:7597
    at wrappedCallback (angular.js:10968)
    at wrappedCallback (angular.js:10968)
    at angular.js:11054
    at Scope.$eval (angular.js:11974)
    at Scope.$digest (angular.js:11800)
    at Scope.$apply (angular.js:12080)
    at done (angular.js:7862)
    at completeRequest (angular.js:8045)console.(anonymous function) @ logbar.js:11(anonymous function) @ angular.js:9438(anonymous function) @ angular.js:6857wrappedCallback @ angular.js:10971wrappedCallback @ angular.js:10968(anonymous function) @ angular.js:11054$eval @ angular.js:11974$digest @ angular.js:11800$apply @ angular.js:12080done @ angular.js:7862completeRequest @ angular.js:8045xhr.onreadystatechange @ angular.js:8001
```
$scope.model[cmp].globalBytes is not always there when the calculation happens.

### PR DESCRIPTION
**Pull request by calmh**
### Purpose

This adds the byte counts to the /rest/db/completion result and FolderCompletion events so the data is directly available for the GUI.

### Testing

I did some manual poking around and looked in the javascript console.

### CODE REVIEW

**Comment by AudriusButkevicius** [comment]
@st-review merge

**Comment by st-review** [comment]
:ok_hand: Merged as 7776839c820134abb8c02e22b2b87c7622339682. Thanks, @calmh!

File: gui/default/syncthing/core/syncthingController.js [hunk 1]
```diff
@@ -458,6 +438,20 @@ angular.module('syncthing.core')
...
+        function recalcCompletion(device) {
+            var total = 0, needed = 0;
+            for (var folder in $scope.completion[device]) {
+                if (folder === "_total") {
+                    continue;
+                }
+                total += $scope.completion[device][folder].globalBytes;
+                needed += $scope.completion[device][folder].needBytes;
+            }
+            $scope.completion[device]._total = 100 * (1 - needed / total);
```
Comment (AudriusButkevicius) [inline gui/default/syncthing/core/syncthingController.js]: Divison by zero...

Comment (AudriusButkevicius) [inline gui/default/syncthing/core/syncthingController.js]: If it shares no folders.

**Comment by AudriusButkevicius** [comment]
Needs a fix for division.

**Comment by calmh** [comment]
If it shares no folders it won't end up in that code, but there may be a folder that is zero bytes large so yes, will fix!

**Pull request closed without merging**